tests/components/telegram_bot/test_telegram_bot.py

```diff
@@ -1648,3 +1698,171 @@ async def test_deprecated_timeout_parameter(
 
     # verify issue is resolved
     assert not issue_registry.async_get_issue(DOMAIN, "deprecated_timeout_parameter")
+
+
+@pytest.mark.parametrize(
+    (
+        "telegram_file_name",
+        "schema_request",
+        "expected_download_path",
+        "expected_allowlist_dir",
+    ),
+    [
+        pytest.param(
+            "file_name1.jpg",
+            {ATTR_FILE_ID: "some-file-id-1"},
+            lambda hass: f"{hass.config.path(DOMAIN)}/file_name1.jpg",
+            lambda hass: hass.config.path(DOMAIN),
+            id="no_custom_name_no_custom_dir",
+        ),
+        pytest.param(
+            "file_name2.jpg",
+            {ATTR_FILE_ID: "some-file-id-2", ATTR_FILE_NAME: "custom_name2.jpg"},
+            lambda hass: f"{hass.config.path(DOMAIN)}/custom_name2.jpg",
+            lambda hass: hass.config.path(DOMAIN),
+            id="custom_name_no_custom_dir",
+        ),
+        pytest.param(
+            "file_name3.jpg",
+            {
+                ATTR_FILE_ID: "some-file-id-3",
+                ATTR_DIRECTORY_PATH: "/tmp/tempfolder3",  # noqa: S108
+            },
+            lambda hass: "/tmp/tempfolder3/file_name3.jpg",  # noqa: S108
+            lambda hass: "/tmp/tempfolder3",  # noqa: S108
+            id="no_custom_name_custom_dir",
+        ),
+        pytest.param(
+            "file_name4.jpg",
+            {
+                ATTR_FILE_ID: "some-file-id-4",
+                ATTR_FILE_NAME: "custom_name4.jpg",
+                ATTR_DIRECTORY_PATH: "/tmp/tempfolder4",  # noqa: S108
+            },
+            lambda hass: "/tmp/tempfolder4/custom_name4.jpg",  # noqa: S108
```

Comment:
Shouldn't we use the tmp_path fixture here too? I'm hesitant about writing to the file system in other ways.

Reply:
I thought it would be nice to have all the cases in the same test with parameterize. If I'll use the tmp_path I will need to split the test. 
What do u think?

P.S.
We do use /tmp path explicitly in other tests